src/test/java/org/takes/misc/VerboseIteratorTest.java

```diff
@@ -0,0 +1,72 @@
+package org.takes.misc;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.collection.IsIterableWithSize;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * 
+ * @author marcus.sanchez (sanchez.marcus@gmail.com)
+ * @version $Id$
+ * @since 0.15.1
+ */
+public class VerboseIteratorTest {
+
+	private static final String NO_NEXT_VALUE_ERROR = "Empty Error Message";
+
+	private final List<String> VALID_LIST = Arrays.asList("Accept: text/plain",
+			"Accept-Charset: utf-8",
+			"Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==",
+			"Cache-Control: no-cache", "From: user@example.com");
+
+	private final List<String> EMPTY_LIST = Arrays.asList();
+
+	@Rule
+	public final ExpectedException thrown = ExpectedException.none();
+
+	@Test
```

Comment:
@m4solutions every method needs a javadoc. Test methods should follow this http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names
I suggest `VerboseIterator can return the next value` along with a method name of `returnsNextValue()`.